service/frontend/workflow_handler.go

```diff
@@ -4988,3 +4926,157 @@ func (wh *WorkflowHandler) unaliasCreateScheduleRequestSearchAttributes(request
 }
+
+func (wh *WorkflowHandler) getScheduleExecutionInfoAndArgs(ctx context.Context, workflowID string, namespaceName string, namespaceID namespace.ID) (*workflowpb.WorkflowExecutionInfo, *schedspb.StartScheduleArgs, error) {
+	execution := &commonpb.WorkflowExecution{WorkflowId: workflowID}
+	// describe to get memo, search attributes, and execution info
+	describeResponse, err := wh.historyClient.DescribeWorkflowExecution(ctx, &historyservice.DescribeWorkflowExecutionRequest{
+		NamespaceId: namespaceID.String(),
+		Request: &workflowservice.DescribeWorkflowExecutionRequest{
+			Namespace: namespaceName,
+			Execution: execution,
+		},
+	})
+	if err != nil {
+		// TODO: rewrite "workflow" in error messages to "schedule"
+		return nil, nil, err
+	}
+
+	executionInfo := describeResponse.GetWorkflowExecutionInfo()
+	var startScheduleArgs schedspb.StartScheduleArgs
+	switch executionInfo.GetStatus() {
+	case enumspb.WORKFLOW_EXECUTION_STATUS_COMPLETED:
+		reverseHistory, err := wh.GetWorkflowExecutionHistoryReverse(ctx, &workflowservice.GetWorkflowExecutionHistoryReverseRequest{
+			Namespace:       namespaceName,
+			Execution:       &commonpb.WorkflowExecution{WorkflowId: workflowID, RunId: executionInfo.Execution.RunId},
+			MaximumPageSize: 1,
+		})
+		if err != nil {
+			return nil, nil, err
+		}
+		events := reverseHistory.History.GetEvents()
+		if len(events) == 0 || events[0].EventType != enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_COMPLETED || events[0].GetWorkflowExecutionCompletedEventAttributes() == nil {
+			return nil, nil, serviceerror.NewInternal("schedule result cannot be loaded")
+
+		}
+		err = sdk.PreferProtoDataConverter.FromPayloads(events[0].GetWorkflowExecutionCompletedEventAttributes().Result, &startScheduleArgs)
+		if err != nil {
+			return nil, nil, err
+		}
+
+	case enumspb.WORKFLOW_EXECUTION_STATUS_RUNNING:
+		// leave startScheduleArgs empty
+	default:
+		return nil, nil, serviceerror.NewNotFound("schedule not found")
+	}
+	return executionInfo, &startScheduleArgs, nil
+}
+
+func (wh *WorkflowHandler) checkEventBlobSizeLimit(ctx context.Context, workflowID string, namespaceName string, namespaceID namespace.ID, size int, operationName string) error {
+	sizeLimitError := wh.config.BlobSizeLimitError(namespaceName)
+	sizeLimitWarn := wh.config.BlobSizeLimitWarn(namespaceName)
+	return common.CheckEventBlobSizeLimit(
+		size,
+		sizeLimitWarn,
+		sizeLimitError,
+		namespaceID.String(),
+		workflowID,
+		"", // don't have runid yet
+		wh.metricsScope(ctx).WithTags(metrics.CommandTypeTag(enumspb.COMMAND_TYPE_UNSPECIFIED.String())),
+		wh.throttledLogger,
+		tag.BlobSizeViolationOperation(operationName),
+	)
+}
+
+func (wh *WorkflowHandler) signalWithStartScheduleWorkflow(ctx context.Context, workflowID string, namespaceName string, signalName string, identity string, requestId string, signalInputPayloads *commonpb.Payloads, getStartInputPayloads func(*schedspb.StartScheduleArgs) (*commonpb.Payloads, error)) error {
```

Comment:
change the name to something else, since it's not doing exactly SignalWithStart anymore?